.gitignore

```diff
@@ -21,3 +29,31 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+### STS ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+
+### IntelliJ IDEA ###
+.idea
+*.iws
+*.iml
+*.ipr
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+build/
+!**/src/main/**/build/
+!**/src/test/**/build/
```

Comment:
Should Netbeans gitignore be just `/` or `**/`?

There is an existing project taking the guesswork out of things like this away. https://github.com/github/gitignore

I think one thing that can be kept in mind is to think:

- How is it being done by others?
- And why are they doing it that way?
- Why is it different from the way I am doing it?

Reply:
Thank you, Sir. I think it is a pretty good standard to follow. It's also a refresher for me to review our internship project and other spring projects. I have also noticed different styles and other directories/files that should be put in the gitignore but we failed to do so. I will definitely change this in my next commit.